ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -14,11 +14,82 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import argparse
+import os
 import sys
+import time
+
+from ros2_generate_interface_docs import msg_utils
+from ros2_generate_interface_docs import utils
+
+from rosidl_runtime_py import get_message_interfaces
+
+
+def generate_interfaces(interfaces, html_dir, template, interface_type):
+    """
+    Generate the index and documentation for each message.
+
+    Parameters
+    ----------
+    interfaces: dict of {str : str[]}
+        dictionary with the interface package name associated with all the interface
+        for this package.
+    html_dir: str
+        path to the directory to save the generated documentation
+    template: str
+        template name
+    interface_type: str
+        type of the interface: msg, action or srv
+
+    """
+    for package_name, interface_names in sorted(interfaces.items(), key=lambda item: item[0]):
+        package_directory = os.path.join(html_dir, package_name)
+        interface_type_directory = os.path.join(package_directory, interface_type)
+        os.makedirs(interface_type_directory, exist_ok=True)
+        utils.generate_index(package_name,
+                             package_directory,
+                             interfaces[package_name])
+        for interface_name in interface_names:
+            documentation_data = {'name': interface_name,
+                                  'package': package_name,
+                                  'base_type': interface_name,
+                                  'date': str(time.strftime('%a, %d %b %Y %H:%M:%S'))}
+
+            if(interface_type == 'msg'):
+                documentation_data = {**documentation_data,
+                                      **{'ext': 'msg', 'type': 'Message'}}
+                function_to_generate_text_from_spec = msg_utils.generate_msg_text_from_spec
+
+            utils.generate_interface_documentation('%s/%s' % (package_name, interface_name),
+                                                   template,
+                                                   os.path.join(package_directory,
+                                                                interface_name + '.html'),
+                                                   documentation_data,
+                                                   function_to_generate_text_from_spec)
 
 
 def main(argv=sys.argv[1:]):
-    print('Hello ros2_generate_interface_docs')
+    """Parse the arguments and call the right logic."""
+    parser = argparse.ArgumentParser(
+        description='Generate interfaces public API documentation',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '--outputdir', type=str, default='api',
+        help='Output directory')
+    args = parser.parse_args(argv)
+
+    output_dir = args.outputdir
+    os.makedirs(output_dir, exist_ok=True)
+    html_dir = os.path.join(output_dir, 'html')
```

Comment:
Nitpick: by using `args.outputdir` in this line the previous two lines are unnecessary.